File: VB/DateNavigatorCustomized/MyDateNavigator.vb
Imports DevExpress.XtraScheduler
Imports System

Namespace DateNavigatorCustomized
    #Region "#mydatenavigator"
    Public Class MyDateNavigator
        Inherits DateNavigator

        Protected Overrides Function AdjustSelectionEnd(ByVal start As Date, ByVal [end] As Date) As Date
            Return [end]
            'return base.AdjustSelectionEnd(start, end);
        End Function

        Protected Overrides Function AdjustSelectionStart(ByVal start As Date, ByVal [end] As Date) As Date
            Return start
            'return base.AdjustSelectionStart(start, end);
        End Function
    End Class
    #End Region ' #mydatenavigator
End Namespace

File: CS/DateNavigatorCustomized/MyDateNavigator.cs
using DevExpress.XtraScheduler;
using System;

namespace DateNavigatorCustomized
{
    #region #mydatenavigator
    public class MyDateNavigator : DateNavigator
    {
        protected override DateTime AdjustSelectionEnd(DateTime start, DateTime end)
        {
            return end;
            //return base.AdjustSelectionEnd(start, end);
        }

        protected override DateTime AdjustSelectionStart(DateTime start, DateTime end)
        {
            return start;
            //return base.AdjustSelectionStart(start, end);
        }
    }
    #endregion #mydatenavigator
}
